### PR TITLE
Mini-challenge 3 - Hooks

### PR DESCRIPTION
**Pull request by Kingarturs**
### New features:
- VideoDetails page
- Video Search Capability
- Suggestion of videos based on the Youtube API
- Render of search results on home view
- Loading of any video putting the id on the browser

### Test coverage:
I found it difficult to make the tests for this specific challenge, specially with the approach i decided to use in order to load the video details, i need some more practice and a lot more experience using Jest, not all my tests pass because of the way i made them, i think is caused by a memory leak that doesn't let the `useYoutubeApi` custom hook to set the state i need for the videos i test to load on the `VideoDetails.test.js` file.

Anyways, I'll keep looking into it.
Here's the test coverage screenshot

image

### CODE REVIEW

**Comment by Kingarturs** [comment]
I didn't forgot 😁, i still have to make some changes:
- The responsiveness on the navbar that needs improvements on small screens (I'm adding a hamburger menu).
- The tests for the App component.

I'm adding them on the next PR 🚀.

File: src/hooks/useDebounce.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { useRef, useEffect } from 'react';
```
Comment (andresSaldanaAguilar) [inline src/hooks/useDebounce.js:1]: Nice implementation 😄

File: src/pages/Home/Home.page.jsx [hunk 2]
```diff
@@ -1,80 +1,31 @@
...
-                      : 'No description'}
-                  </p>
-                </div>
-              </VideoCard>
-            );
-          }
-
-          return null;
-        })}
+      <NavBar search setSearch={setSearch} />
```
Comment (andresSaldanaAguilar) [inline src/pages/Home/Home.page.jsx:20]: you can avoid adding the navbar to every view creating a context for the search as an idea, you addressed pretty well this, congrats!

File: src/providers/Theme/Theme.provider.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import React, { useState, useContext, useCallback } from 'react';
```
Comment (andresSaldanaAguilar) [inline src/providers/Theme/Theme.provider.jsx:1]: cool, nice implementation for providing context!

File: src/hooks/useYoutubeApi.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
+import { useState } from 'react';
+import axios from 'axios';
+
+const useYoutubeApi = () => {
```
Comment (andresSaldanaAguilar) [inline src/hooks/useYoutubeApi.js:4]: You can use a `reducer` to dispatch every case you're handling

**Comment by andresSaldanaAguilar** [comment]
Good job Luis! I dare to say that your interface is my favorite, you took attention to user experience and design 🥇 , your custom hooks are nice, I just leave some suggestions for you to take more advantage of context, and also to take advantage of reducers on the API hook, and don't worry, tests can be difficult at the beginning. If you need some help please slack me and we can do a 1:1 😄 .

## Acceptance Criteria
- [x] Videos in the Home View are fetched from the YouTube API and displayed correctly.
- [x] Users can search for YouTube videos using the search field on the header.
- [x] A video can be played from the **Video Details View** after clicking on it.
- [x] The video information and related videos list are displayed correctly on the **Video Details View**.
- [x] When a user clicks on a related video the video player, information, and related videos list are updated.

### Bonus Points
- [ ] Custom Hooks for API calls are implemented and tested correctly - Couldn't find tests for hooks.
- [ ] Testing coverage is above 60%. Coverage is high but I could find as many tests as expected for 60% of coverage.

**Pull request merged**